pkg/operator/controller/ingress/controller.go

```diff
@@ -134,9 +144,51 @@ func New(mgr manager.Manager, config Config) (controller.Controller, error) {
 	if err := c.Watch(source.Kind(operatorCache, &configv1.Proxy{}), handler.EnqueueRequestsFromMapFunc(reconciler.ingressConfigToIngressController)); err != nil {
 		return nil, err
 	}
+	// Watch secrets in the operand namespace that are referenced as default certificates so that
+	// checkDefaultCertificate can determine whether changes to the default certificate affect upgradeability.
+	if err := c.Watch(source.Kind(operatorCache, &corev1.Secret{}), handler.EnqueueRequestsFromMapFunc(reconciler.secretToIngressController), predicate.Funcs{
+		CreateFunc: func(e event.CreateEvent) bool {
+			return e.Object.GetNamespace() == operatorcontroller.DefaultOperandNamespace
+		},
+		DeleteFunc: func(e event.DeleteEvent) bool {
+			return e.Object.GetNamespace() == operatorcontroller.DefaultOperandNamespace
+		},
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			return e.ObjectNew.GetNamespace() == operatorcontroller.DefaultOperandNamespace
+		},
+		GenericFunc: func(e event.GenericEvent) bool { return false },
+	}); err != nil {
+		return nil, err
+	}
 	return c, nil
 }
 
+// secretToIngressController maps a secret to a slice of reconcile requests,
+// one request per IngressController that references the secret as it's default certificate.
```

Comment:
```suggestion
// one request per IngressController that references the secret as its default certificate.
```
But this isn't worth worrying about if you aren't making any other changes.

Reply:
Done.